### PR TITLE
Texture: fix destruction and uninitialized member access

### PR DESCRIPTION
**Pull request by hjanetzek**
if m_validData was accidental 0 font atlas were constantly re-uploaded

### CODE REVIEW

**Comment by karimnaaji** [comment]
LGTM

**Comment by tallytalwar** [comment]
This is good. And some nice catches.

File: core/src/gl/mesh.cpp [hunk 1]
```diff
@@ -37,17 +37,21 @@ MeshBase::~MeshBase() {
     // Deleting a index/array buffer being used ends up setting up the current vertex/index buffer to 0
     // after the driver finishes using it, force the render state to be 0 for vertex/index buffer
 
-    if (m_glVertexBuffer) {
-        if (RenderState::vertexBuffer.compare(m_glVertexBuffer)) {
-            RenderState::vertexBuffer.init(0, false);
+    if (RenderState::isValidGeneration(m_generation)) {
```
Comment (karimnaaji) [inline core/src/gl/mesh.cpp]: Do you think we should also defer those in main thread loop?

Comment (tallytalwar) [inline core/src/gl/mesh.cpp]: isnt Mesh destructor already invoked from mainthread? (tileWorker does that?)

Comment (hjanetzek) [inline core/src/gl/mesh.cpp]: Yes, only textures were a special case for the caches in RasterSource so far. At some point we could have a GLResource class that holds the gl handle and moves things for destruction to the mainloop unless it's called on the mainloop already.

Comment (tallytalwar) [inline core/src/gl/mesh.cpp]: Making more work for @blair1618 to rebase/redo!
Refer: https://github.com/tangrams/tangram-es/commit/ba0dcf9bb4f94a383d61aff036cf98107a3308cf

**Pull request merged**